moses/atomese/representation/atomese_representation.cc

```diff
@@ -0,0 +1,444 @@
+{
+	return createLink(EXECUTION_LINK,
+	                  createNode(SCHEMA_NODE, label),
+	                  mk_row_number_cell(row_num),
+	                  mk_real_cell(contin));
+}
+
+/**
+ * Create a True / False link based on a given builtin value.
+ *
+ * @param b builtin value
+ * @return
+ */
+Handle mk_boolean_cell(const builtin& b)
+{
+	return createLink(builtin_to_bool(b) ? TRUE_LINK : FALSE_LINK);
+}
+
+/**
+ * Represent a real cell as an Evaluation of the corresponding feature
+ * on its row (i.e Predicate of feature on Node of row).
+ *
+ * @param b builtin value
+ * @param row_num row number
+ * @param label label of the feature
+ * @return
+ */
+Handle mk_boolean_cell_eval(const builtin& b,
+                            const int& row_num, const string& label)
+{
+	Handle h = createLink(EVALUATION_LINK,
+	                      createNode(PREDICATE_NODE, label),
+	                      mk_row_number_cell(row_num));
+	TruthValuePtr tv(SimpleTruthValue::createTV(builtin_to_bool(b), 1));
+	h->setTruthValue(tv);
+	return h;
+}
+
+/**
+ * Represent a boolean cell as an Execution of the corresponding feature
+ * on its row (i.e Schema of feature on Node of row).
+ *
+ * @param b builtin value
+ * @param row_num row number
+ * @param label label of the feature
+ * @return
+ */
+Handle mk_boolean_cell_exec(const builtin& b,
+                            const int& row_num, const string& label)
+{
+	return createLink(EXECUTION_LINK,
+	                  createNode(SCHEMA_NODE, label),
+	                  mk_row_number_cell(row_num),
+	                  mk_boolean_cell(b));
+}
+
+/**
+ * Represent a row as a list of cells.
+ *
+ * @param cells cells in a row
+ * @return
+ */
+Handle mk_row(const HandleSeq& cells)
+{
+	return createLink(cells, LIST_LINK);
+}
+
+/**
+ * Represent a row as a list of row number and cells
+ *
+ * @param cells cells in a row
+ * @param row_num row number
+ * @return
+ */
+Handle mk_numbered_row(const HandleSeq& cells, const int& row_num)
+{
+	return createLink(LIST_LINK,
+	                  mk_row_number_cell(row_num),
+	                  createLink(cells, LIST_LINK));
+}
+
+/**
+ * Create a table of dependent (output) data.
+ *
+ * @param oTable output table
+ * @return
+ */
+Handle mk_output_table(const OTable& oTable)
+{
+	const type_node& type = oTable.get_type();
+	const string& label = oTable.get_label();
+	HandleSeq rows;
+	int row_num = 1;
+
+	if (type == id::boolean_type)
+	{
+		for (const vertex& v : oTable)
+		{
+			Handle h = createLink(LIST_LINK,
+			                      mk_row_number_cell(row_num++),
+			                      mk_boolean_cell(boost::get<builtin>(v)));
+			rows.push_back(h);
+		}
+	}
+	else if (type == id::contin_type)
+	{
+		for (const vertex& v : oTable)
+		{
+			Handle h = createLink(LIST_LINK,
+			                      mk_row_number_cell(row_num++),
+			                      mk_real_cell(boost::get<contin_t>(v)));
+			rows.push_back(h);
+		}
+	}
```

Comment:
Alternatively you can put the conditional in the `createLink` and select either `mk_real_cell` or `mk_boolean_cell` depending on `type`. It slightly less efficient (though probably imperceptibly) but it reduces the number of lines of code.

Another even better option is to create a `mk_cell` visitor! that way you just call that `mk_cell` function on `v` and you're good. There are plenty of examples of visitors throughout MOSES' code (in `table.h` for instance). See also https://www.boost.org/doc/libs/1_67_0/doc/html/variant.html for its documentation.

Reply:
>  It slightly less efficient (though probably imperceptibly) but it reduces the number of lines of code.

Will it be imperceptible as we're going to do the checking for every single cell in the data?

Reply:
In fact, if we're changing it for only the output table, it wouldn't affect much.